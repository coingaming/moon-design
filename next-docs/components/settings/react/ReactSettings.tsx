import { useState } from 'react';
import {
  LogoSportsbetShort,
  LogoBitcasinoShort,
  LogoMoonDesignShort,
  LogoBetaddaShort,
} from '@heathmont/moon-assets';
import useFeatureFlags from '../../../utils/useFeatureFlags';
import { useDocsTheme, DocsBrands } from '../../themes/DocsThemeProvider';
import useThemeTW from '../../themes/useThemesTW';
import ReactElixirSwitcher from '../common/ReactElixirSwitcher';
import SettingsButton from '../common/SettingsButton';
import DarkLightModeSwitcher from './DarkLightModeSwitch';
import RTLModeSwitch from './RTLModeSwitch';

type LogosKeys = keyof typeof Logos;

interface Props {
  toggleReactAndElixir: () => void;
  isReactEnabled: boolean;
}

const Logos = {
  sportsbet: <LogoSportsbetShort fontSize="1rem" />,
  bitcasino: <LogoBitcasinoShort fontSize="1rem" />,
  moonDesign: <LogoMoonDesignShort fontSize="1rem" />,
  betadda: <LogoBetaddaShort fontSize="1rem" />,
};

const ReactSettings: React.FC<Props> = ({
  toggleReactAndElixir,
  isReactEnabled,
}) => {
  const [isOpened, setIsOpened] = useState(false);
  const toggle = () => setIsOpened(!isOpened);

  const { toggleColorScheme, setBrand, themeKeys } = useDocsTheme();

  const [isRtlEnabled, setIsRtl] = useState(false);

  const { isFeatureElixirEnabled } = useFeatureFlags();

  const toggleRtl = () => {
    const htmlTag = document && document?.getElementsByTagName('html')[0];
    setIsRtl(!isRtlEnabled);
    const dirAttr = htmlTag.getAttribute('dir');

    if (!dirAttr) {
      htmlTag.setAttribute('dir', 'rtl');
    } else {
      htmlTag.removeAttribute('dir');
    }
  };

  const filteredThemeKeys = themeKeys.filter((themeKey: string) =>
    ['sportsbet', 'bitcasino', 'moonDesign', 'betadda'].includes(themeKey)
  );

  const {
    setBrand: setBrandTW,
    toggleMode: toggleModeTW,
    getMode,
  } = useThemeTW();

  const switchBrandHandler = (themeKey: LogosKeys) => {
    setBrand(themeKey as DocsBrands);
    setBrandTW(themeKey);
  };

  const switchModeHandler = () => {
    toggleColorScheme();
    toggleModeTW();
  };
  return (
    <>
      <SettingsButton onClick={toggle} isRtlEnabled={isRtlEnabled} />
      {isOpened && (
        <>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-16 z-50`}
          >
            {filteredThemeKeys.map((themeKey: string) => {
              const Logo = Logos[themeKey as LogosKeys] || Logos.moonDesign;
              return (
                <button
                  key={themeKey}
                  type="button"
                  onClick={() => switchBrandHandler(themeKey as LogosKeys)}
                  className={`${
                    isRtlEnabled ? `mr-4` : `ml-4`
                  } p-2 rounded-full text-black bg-white hover:bg-gray-200 inline-flex items-center justify-center shadow-md`}
                >
                  {Logo}
                </button>
              );
            })}
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-28 z-50`}
          >
            <DarkLightModeSwitcher
              toggle={switchModeHandler}
              isEnabled={getMode() === 'dark'}
              isRtlEnabled={isRtlEnabled}
            />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-40 z-50`}
          >
            <RTLModeSwitch toggle={toggleRtl} isEnabled={isRtlEnabled} />
          </div>
          <div
            className={`${
              isRtlEnabled ? `left-4` : `right-4`
            } fixed bottom-52 z-50`}
          >
            {isFeatureElixirEnabled ? (
              <ReactElixirSwitcher
                toggle={toggleReactAndElixir}
                isReactEnabled={isReactEnabled}
              />
            ) : null}
          </div>
        </>
      )}
    </>
  );
};

export default ReactSettings;
