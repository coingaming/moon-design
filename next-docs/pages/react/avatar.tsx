import React from 'react';

import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageAvatar() {
  return (
    <>
      <ThemeProvider theme={sportsbetDark}>
        <h1>Avatar</h1>
      </ThemeProvider>
    </>
  );
}
