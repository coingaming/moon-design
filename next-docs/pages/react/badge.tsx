import React from 'react';
import Head from 'next/head';

import { Badge } from '@heathmont/moon-core';
import { ThemeProvider, sportsbetDark } from '@heathmont/moon-themes';

export default function PageBadge() {
  return (
    <>
      <h1>Badge</h1>
      <p>Small count and labeling component.</p>
      <Badge>Active</Badge>
      <Badge color="goten.100" backgroundColor="piccolo.100">
        Active
      </Badge>
    </>
  );
}
