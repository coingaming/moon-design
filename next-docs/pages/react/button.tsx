import React from 'react';
import Head from 'next/head';

import { Button } from '@heathmont/moon-components';
import PreviewSwitch from '../../components/codePreview/Switch';
import Preview from '../../components/codePreview/Preview';

export default function PageBadge() {
  return (
    <>
      <section className="mt-6">
        <h1 className="text-5xl font-semibold">Button</h1>
        <p className="text-lg mt-4">
          Buttons are calls-to-action used to prompt users. They encourage users
          to interact with us in multiple ways throughout our galaxy, based on
          what the label of the button indicates.
        </p>

        <p className="text-lg mt-4">
          Buttons are clickable elements with label text that describe the
          action that will happen when the users interact with it.
        </p>
      </section>
      <section className="mt-6">
        <Preview
          title="Primary"
          preview={<Button variant="primary">Active</Button>}
          code={`import { Button } from '@heathmont/moon-components';

<Button variant="primary">Active</Button>`
}
        />
      </section>
    </>
  );
}
