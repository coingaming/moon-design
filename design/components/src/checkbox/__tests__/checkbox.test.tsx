import * as React from 'react';
import { create } from 'react-test-renderer';
import { Checkbox } from '../';

const testProps = {
  label: 'I agree to the Terms & Conditions',
};

describe('Checkbox', () => {
  test('renders by default', () => {
    const checkbox = create(<Checkbox {...testProps} />);

    expect(checkbox).toMatchSnapshot();
  });

  test('renders as disabled', () => {
    const checkbox = create(<Checkbox disabled {...testProps} />);

    expect(checkbox).toMatchSnapshot();
  });
});
