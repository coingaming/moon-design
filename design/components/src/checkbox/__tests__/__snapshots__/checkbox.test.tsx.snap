// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders as disabled 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  height: 1.25rem;
  width: 1.25rem;
  position: relative;
  margin-right: 1rem;
  padding: 0.125rem;
  background-clip: content-box;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: 1px solid #31373F;
  border-radius: 4px;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-0:checked {
  background-color: #0cd664;
}

.emotion-0:disabled {
  border-color: #304854;
  cursor: not-allowed;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #304854;
}

<div
  className="emotion-2"
>
  <input
    aria-label="I agree to the Terms & Conditions"
    className="emotion-0"
    disabled={true}
    id="Checkbox-i-agree-to-the-terms-conditions"
    type="checkbox"
  />
  <label
    className="emotion-1"
    htmlFor="Checkbox-i-agree-to-the-terms-conditions"
  >
    I agree to the Terms & Conditions
  </label>
</div>
`;

exports[`Checkbox renders by default 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  height: 1.25rem;
  width: 1.25rem;
  position: relative;
  margin-right: 1rem;
  padding: 0.125rem;
  background-clip: content-box;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: 1px solid #31373F;
  border-radius: 4px;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-0:checked {
  background-color: #0cd664;
}

.emotion-0:disabled {
  border-color: #304854;
  cursor: not-allowed;
}

.emotion-1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  color: #8697a2;
}

<div
  className="emotion-2"
>
  <input
    aria-label="I agree to the Terms & Conditions"
    className="emotion-0"
    disabled={false}
    id="Checkbox-i-agree-to-the-terms-conditions"
    type="checkbox"
  />
  <label
    className="emotion-1"
    htmlFor="Checkbox-i-agree-to-the-terms-conditions"
  >
    I agree to the Terms & Conditions
  </label>
</div>
`;

exports[`Checkbox renders with hidden label 1`] = `
.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  height: 1.25rem;
  width: 1.25rem;
  position: relative;
  margin-right: 1rem;
  padding: 0.125rem;
  background-clip: content-box;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  border: 1px solid #31373F;
  border-radius: 4px;
  background-color: transparent;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.emotion-0:checked {
  background-color: #0cd664;
}

.emotion-0:disabled {
  border-color: #304854;
  cursor: not-allowed;
}

<div
  className="emotion-1"
>
  <input
    aria-label="I agree to the Terms & Conditions"
    className="emotion-0"
    disabled={false}
    id="Checkbox-i-agree-to-the-terms-conditions"
    type="checkbox"
  />
</div>
`;
